libafl_bolts/src/lib.rs

```diff
@@ -341,6 +341,14 @@ pub enum Error {
     InvalidCorpus(String, ErrorBacktrace),
     /// Error specific to a runtime like QEMU or Frida
     Runtime(String, ErrorBacktrace),
+    /// Configuration errors, such as incorrect map sizes
+    Configuration(String, ErrorBacktrace),
+    /// Compilation-related errors
+    Compilation(String, ErrorBacktrace),
+    /// System call or resource allocation failures
+    System(String, ErrorBacktrace),
```

Comment:
This is almost always identical to OsError, right? Maybe make OsError's io::Error be optional, instead